src/Tools/DynamoShapeManager/Utilities.cs

```diff
@@ -308,10 +312,15 @@ public static void PreloadAsmFromPath(string preloaderLocation, string asmLocati
                 throw new MissingMethodException(
                     string.Format("Method '{0}' not found", PreloaderMethodName));
             }
-
-            var methodParams = new object[] { asmLocation };
-            preloadMethod.Invoke(null, methodParams);
-
+            try
+            {
+                var methodParams = new object[] { asmLocation };
+                preloadMethod.Invoke(null, methodParams);
+            }
+            catch
+            {
+                Console.WriteLine("Failed to load ASM binaries");
```

Comment:
Won't the exception be silently swallowed out here? I don't say a throw?

Reply:
yes, okay, after discussing a bit I see what you're saying - I think instead of Dynamo starting with no geometry - it is likely more useful to throw and message to the user where the geometry library failed to load from - this will also help us later get to the bottom of crash reports without needing to recreate the users local adsk install setup.

Reply:
I guess I'm okay with launching Dynamo without the geometry library but in addition to doing that I think it would be useful to display a message to users saying that the geometry library failed to load due to these reasons but if they wish they can still continue to use Dynamo.

Reply:
I guess I meant to ask if it's possible to show the message box, the user clicks ok, and is able to continue to use Dynamo?

Reply:
I don't think so - not without a larger refactoring - but I can look at that if you think it's worth it.

Reply:
or not without multiple potential message boxes being shown.

Reply:
hmm - for some context, we should not introduce a dependency on `system.windows` into the preloader - which needs to be cross platform - but the other message box raised in dynamoSandbox only is shown when it is too late to recover.

Reply:
@aparajit-pratap PTAL - figured out a way to do this while avoiding big set of changes or new public apis. We may want to do that as part of the followup we discussed.